core/ledger/kvledger/tests/reset_test.go

```diff
@@ -217,3 +218,49 @@ func TestResetLedgerWithoutDroppingDBs(t *testing.T) {
 		"This is possible when the state database is not dropped after a ledger reset/rollback. "+
 		"The state database can safely be dropped and will be rebuilt up to block store height upon the next peer start")
 }
+
+func TestResetFailIfAnyLedgerBootstrappedFromSnapshot(t *testing.T) {
+	env := newEnv(t)
+	defer env.cleanup()
+	env.initLedgerMgmt()
+
+	// populate ledgers with sample data
+	ledgerID := "testLedgerFromSnapshot"
+	dataHelper := newSampleDataHelper(t)
+	h := env.newTestHelperCreateLgr(ledgerID, t)
+	dataHelper.populateLedger(h)
+	dataHelper.verifyLedgerContent(h)
+	bcInfo, err := h.lgr.GetBlockchainInfo()
+	require.NoError(t, err)
+
+	// create a sanapshot
+	blockNum := bcInfo.Height - 1
+	require.NoError(t, h.lgr.SubmitSnapshotRequest(blockNum))
+	// wait until snapshot is generated
+	snapshotGenerated := func() bool {
+		requests, err := h.lgr.PendingSnapshotRequests()
+		require.NoError(t, err)
+		return len(requests) == 0
+	}
+	require.Eventually(t, snapshotGenerated, time.Minute, 100*time.Millisecond)
```

Comment:
100ms seem unnecessarily long to me...

Reply:
The sample ledger has 9 blocks. When running UT locally, there were the following timestamps, so the elapsed time was 337-202=135ms. 

2020-10-09 16:21:59.202 EDT [kvledger] func2 -> INFO 1dd^[[0m Generating snapshot channelID=testLedgerFromSnapshot lastCommittedBlockNumber=8

2020-10-09 16:21:59.337 EDT [kvledger] func2 -> INFO 200^[[0m Generated snapshot channelID=testLedgerFromSnapshot lastCommittedBlockNumber=8